### ISSUE TITLE
B A T C H   R E C U R S I O N

### ISSUE THREAD

**Issue by andregarkin** [custom-dev-tools/WampServer-SSL-Auto-Config#1]
Tried to use it, I get a message about the existing directory. 
Please investigate this issue, I am ready to provide any details about my configuration at any time.

  WampServer SSL Auto Config (v1.1.0)
  -----------------------------------

  20:07:47,78 : Parsed configuration file.
  20:07:47,78 : Common 'logs' directory already exists.
  20:07:47,80 : Common 'vhosts\https' directory already exists.
The system cannot find the path specified.
  20:07:47,81 : Common 'vhosts\https' directory already exists.
The system cannot find the path specified.
  20:07:47,83 : Common 'vhosts\https' directory already exists.
The system cannot find the path specified.
  20:07:47,84 : Common 'vhosts\https' directory already exists.
The system cannot find the path specified.
.......
.......
.......
.......
.......
...
The system cannot find the path specified.
  19:57:20,64 : Common 'vhosts\https' directory already exists.
The system cannot find the path specified.
  19:57:20,66 : Common 'vhosts\https' directory already exists.
The system cannot find the path specified.
******  B A T C H   R E C U R S I O N  exceeds STACK limits ******
Recursion Count=312, Stack Usage=90 percent
******       B A T C H   PROCESSING IS   A B O R T E D      ******

Wampserver Configurating Report: 
d:\wamp64\wampConfReport.txt

------ Wampserver configuration report
Tue, 04 Aug 2020 19:07:50 +0000
- Windows NT 10.0 build 18362 (Windows 10)
- Windows Charset: windows-1251
- Wampserver version 3.2.2 - 64bit
- Wampserver install version 3.2.0
- Install directory: d:/wamp64
- Default browser: C:/Program Files (x86)/Google/Chrome/Application/chrome.exe
- Default text editor: C:/Program Files/Notepad++/notepad++.exe
- Default log viewer: C:/Program Files/Notepad++/notepad++.exe
- Apache 2.4.41 - Port 80
- Additional Apache listening ports:
- PHP 7.3.12
- MariaDB 10.4.10 Port 3306
MariaDB basedir = d:/wamp64/bin/mariadb/mariadb10.4.10
MariaDB datadir = d:/wamp64/bin/mariadb/mariadb10.4.10/data
- PHP 5.6.40 for CLI (Internal Wampserver PHP scripts)
- PhpMyAdmin 4.9.2 - Adminer 4.7.5
------ Wampserver configuration ------
 AliasSubmenu = on - NotCheckVirtualHost = off
 NotCheckDuplicate = off - VirtualHostSubMenu = on
 HomepageAtStartup = off - VhostAllLocalIp = on
 SupportMySQL = off - SupportMariaDB = on
 ShowphmyadMenu = on - ShowadminerMenu = on
 NotVerifyPATH = off - HostsLinesLimit = 5000
 NotVerifyHosts = off - NotVerifyTLD = off
 AutoCleanLogs = on - AutoCleanLogsMax = 1000
 AutoCleanLogsMin = 50 - AutoCleanTmp = on
 AutoCleanTmpMax = 1000 - CreateSymlink = symlink
 BackupHosts = on - TraceProcess = off

**Comment by custom-dev-tools** [comment]
Hi Andrii

Thanks for the report.

So I can better understand what is going on can you please provide the latest run of your 'ssl_config.log'.

If you prefer a bit more privacy your can email it to me (along with your config.ini file if you like): matthew.rath@customdevtools.com

This will give me a clearer understanding of the script flow and error point. In conjunction with information from your previous comment, we should be able to figure it out.

Thanks.

Matt

**Comment by andregarkin** [comment]
Thanks for your reaction.
I think I should visit github more often. And also set up notifications to respond faster.
I'll think about providing more information, ok.

I can also say that I am surprised that WAMP did not include everything necessary to support https://localhost by default, similar to how it is done in XAMPP.
I studied "how to setup HTTPS on WAMP" on official WAMP forum, investigate answers on stackoverflow, articles on the Internet.
I think it's time for the creators of WAMP to release an assembly where three configuration files are configured as needed and the SSL certificate files for localhost are already decomposed into their folders.

What you are doing looks cool to me, thank you. I am going to send you additional data by email.

**Comment by andregarkin** [comment]
Through interaction with Matt, it was possible to find out the reason why the script did not perform work.

In the ZIP repository, all files are in UNIX line ending (LF) format.

It is important that the ssl_config.bat script must have the Windows CRLF line endings format.

Converting the line endings format is very easy, for example: open it in Notepad++ and choose Edit -> EOL conversion -> Windows.

I just converted the ssl_config.bat script file format.
After that, I ran it again and got an adequate result.

In my opinion, the script does all the claimed work.

I express my deep gratitude to Matt for his help and such a wonderful project.
